### PR TITLE
Doc improvements

### LINKED ISSUE #244: Add doc with common Data Valuation benchmarks / tasks
Just short descriptions and links to the functions implementing / helping with them

* Data removal / addition
* Outlier detection
* Data debugging
* etc.

### LINKED ISSUE #524: Fix broken references and math in valuation module

### LINKED ISSUE #525: Fix broken references and math in influences module

### PR DESCRIPTION
**Pull request by mdbenito**
### Description

This PR fixes a bunch of doc stuff. The best way to look at it is to browse the docs.

### Changes

- Closes #524 
- Closes #525 
- Closes #244 by un-gpt-ing the text, and adding a page for benchmarking tasks. The latter needs fixing / extending for influence functions @schroedk Wanna take over this? I should also use some of the plots from our recent submission.
- Fixes tons of broken links and references
- Adds an example gallery
- Uses section indices (i.e. main pages for each section)
- Uses navigation tabs
- Updates some external links
- Adds some missing types
- Changes how functions, classes are rendered: now just the name as title and the signature as code below
- **Moves PNGs to LFS** <- Remember to `git lfs install`!

### Checklist

- [ ] ~Wrote Unit tests (if necessary)~
- [x] Updated Documentation (if necessary)
- [x] Updated Changelog
- [ ] ~If notebooks were added/changed, added boilerplate cells are tagged with `"tags": ["hide"]` or `"tags": ["hide-input"]`~

### CODE REVIEW

File: requirements-docs.txt [hunk 1]
```diff
@@ -1,19 +1,21 @@
...
 mkdocs-git-revision-date-localized-plugin
-mkdocs-glightbox
-mknotebooks>=0.8.0
-pygments
+mkdocs-glightbox==0.3.7
 mkdocs-literate-nav
-mkdocs-material>=9.5.0
-mkdocs-section-index
+mkdocs-material==9.5.14
+mkdocs-material[imaging]
```
Comment (AnesBenmerzoug) [inline requirements-docs.txt:14]: Why not just?
```suggestion
mkdocs-material[imaging]==9.5.14
```

Comment (mdbenito) [reply requirements-docs.txt:14]: Fair.

File: requirements-docs.txt [hunk 2]
```diff
@@ -1,19 +1,20 @@
-mike
+griffe==0.42.1
+mike==2.0.0
 markdown-captions
 mkdocs==1.5.3
 mkdocstrings[python]>=0.24
```
Comment (AnesBenmerzoug) [inline requirements-docs.txt:5]: For some reason, in my environment this fails if I don't pin the version of `mkdocstrings-python` to a newer version (See related [issue](https://github.com/mkdocstrings/griffe/issues/197))

```suggestion
mkdocstrings[python]>=0.24
mkdocstrings-python==1.9.0
```

File: mkdocs.yml [hunk 3]
```diff
@@ -110,18 +111,23 @@ plugins:
...
             inherited_members: true
             docstring_style: google
             docstring_section_style: spacy
             line_length: 80
             show_bases: true
             members_order: source
             show_submodules: false
-            show_signature_annotations: false
+            separate_signature: true
+            show_signature_annotations: true
```
Comment (AnesBenmerzoug) [inline mkdocs.yml:123]: Changing these options (or at least one of them) affects the highlighting and readability in the API docs.

| Develop | Current Branch | Suggestion |
|---|---|---|
| image | image | image |

Here's the suggestion extra css used for this:

```css
/* Highlight function names in API docs */
span.doc-function-name {
    color: var(--doc-symbol-function-fg-color);
    background-color: var(--md-code-bg-color);
    font-weight: 500;
}

/* Highlight class names in API docs */
span.doc-class-name {
    color: var(--doc-symbol-class-fg-color);
    background-color: var(--md-code-bg-color);
    font-weight: 500;
}

/* Highlight attribute names in API docs */
span.doc-attribute-name {
    color: var(--doc-symbol-attribute-fg-color);
    background-color: var(--md-code-bg-color);
    font-weight: 500;
}
```

File: docs/getting-started/benchmarking.md [hunk 4]
```diff
@@ -0,0 +1,96 @@
+---
```
Comment (AnesBenmerzoug) [inline docs/getting-started/benchmarking.md:1]: I think that it would be nice to have figures for some if not all tasks to make it easier to understand.

Comment (mdbenito) [reply docs/getting-started/benchmarking.md:1]: I know. It's in my to-do. I have figures for all of them, but wanted to get this merged before the weekend

File: mkdocs.yml (hunk 3 above)
Comment (mdbenito) [reply mkdocs.yml:123]: Hmm... I actually like the muted style. I tried to make the headings with more prominent with [show_symbol_type_heading: true](https://mkdocstrings.github.io/python/usage/configuration/headings/#show_symbol_type_heading) but for some reason neither that nor `show_symbol_type_toc` work.

I'm not a huge fan of that many colors, but if your changes are compatible with dark mode and they include all symbol types we can include them if @schroedk agrees

Comment (mdbenito) [reply mkdocs.yml:123]: Scratch that. After updating with the versions you pinned, the symbol headings appear 🎉

Comment (AnesBenmerzoug) [reply mkdocs.yml:123]: Yes, this looks much nicer!

File: docs/getting-started/benchmarking.md (hunk 4 above)
Comment (AnesBenmerzoug) [reply docs/getting-started/benchmarking.md:1]: I see. Then we can leave it for another PR.

**Review by AnesBenmerzoug — APPROVED** [review]

**Pull request merged**